src/main/java/org/openrewrite/java/apache/httpclient5/AddTimeUnitArgument.java

```diff
@@ -68,21 +69,25 @@ public String getDescription() {
     public TreeVisitor<?, ExecutionContext> getVisitor() {
         return new JavaIsoVisitor<ExecutionContext>() {
             final MethodMatcher matcher = new MethodMatcher(methodPattern);
-            final JavaTemplate template = JavaTemplate.builder("#{any(long)}, TimeUnit.#{}")
-                    .contextSensitive()
-                    .javaParser(JavaParser.fromJavaVersion().classpath("httpclient5", "httpcore5"))
-                    .imports("java.util.concurrent.TimeUnit")
-                    .build();
 
             @Override
             public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext executionContext) {
                 J.MethodInvocation m = super.visitMethodInvocation(method, executionContext);
-                if (matcher.matches(method) && method.getArguments().size() == 1) {
+                if (matcher.matches(m)) {
+                    JavaTemplate template = JavaTemplate
+                            .builder(StringUtils.repeat("#{any()}, ", m.getArguments().size()) + "TimeUnit.#{}")
+                            .contextSensitive()
+                            .javaParser(JavaParser.fromJavaVersion().classpath("httpclient5", "httpcore5"))
```

Comment:
Curious to see if this would work without this import; we're already using `contextSensitive`.. Maybe a quick experiment quick drop this requirement and make the recipe more broadly applicable? Unless this results in missing method type information.
```suggestion
```

Reply:
actually, the unit tests of the individual class, in which the parser of the JavaTemplate does not have visibility to the original A class, and does not know their methods, it's missing some method type information...
Without contextSensitive and adding the classpath of httpclient and httpcore, it's also missing proper type attribution in the output of the template for the migration recipe:
 
<img width="1409" alt="Captura de pantalla 2023-08-02 a las 12 25 13" src="https://github.com/openrewrite/rewrite-spring/assets/597231/ae986a0c-9cec-4304-9dab-0de59833ca90">

Reply:
Thanks for checking! Wasn't sure how much that `contextSensitive` pulls in; good to keep this as is then!